clio-server/src/test/scala/org/broadinstitute/clio/server/dataaccess/HttpElasticsearchDAOSpec.scala

```diff
@@ -8,75 +8,37 @@ import com.sksamuel.elastic4s.http.ElasticDsl._
 import com.sksamuel.elastic4s.searches.SearchDefinition
 import io.circe.parser
-import org.apache.http.HttpHost
 import org.apache.http.util.EntityUtils
-import org.broadinstitute.clio.model.{
-  ElasticsearchField,
-  ElasticsearchIndex,
-  ElasticsearchStatusInfo
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.{
+  AutoElasticsearchIndex,
+  ElasticsearchIndex
 }
 import org.elasticsearch.action.support.WriteRequest.RefreshPolicy
 import org.elasticsearch.client.ResponseException
 import org.scalatest._
 
-import scala.concurrent.Future
-import scala.concurrent.duration._
-
 class HttpElasticsearchDAOSpec
-    extends AsyncFlatSpec
+    extends AbstractElasticsearchDAOSpec("HttpElasticsearchDAO")
+    with AsyncFlatSpecLike
     with Matchers
-    with EitherValues
-    with BeforeAndAfterAll
-    with ElasticsearchContainer {
+    with EitherValues {
   behavior of "HttpElasticsearch"
 
-  lazy val httpHost =
-    new HttpHost(elasticsearchContainerIpAddress, elasticsearchPort)
-  lazy val httpElasticsearchDAO = new HttpElasticsearchDAO(Seq(httpHost))
-
-  private val StatusGreen = "green"
-
-  override protected def afterAll(): Unit = {
-    // Not using regular .close within afterAll. The execution context provided by scalatest doesn't seem to run here.
-    httpElasticsearchDAO.closeClient()
-    super.afterAll()
+  it should "initialize" in {
+    initialize()
   }
 
-  it should "wait up to 60s for green status" in {
-    def waitForGreen(count: Int, sleep: Duration): Future[Assertion] = {
-      httpElasticsearchDAO.getClusterStatus flatMap {
-        case ElasticsearchStatusInfo(StatusGreen, 1, 1) =>
-          Future.successful(succeed)
-        case other if count == 0 =>
-          Future.failed(new RuntimeException(s"Timeout with status: $other"))
-        case _ =>
-          Thread.sleep(sleep.toMillis)
-          waitForGreen(count - 1, sleep)
-      }
-    }
+  it should "create an index and update the index field types" in {
+    case class IndexVersion1(bar: String)
+    case class IndexVersion2(foo: Long, bar: String)
 
-    waitForGreen(20, 3.seconds)
-  }
+    val indexVersion1: ElasticsearchIndex[_] =
+      new AutoElasticsearchIndex[IndexVersion1]("test_index_update_type")
+    val indexVersion2: ElasticsearchIndex[_] =
+      new AutoElasticsearchIndex[IndexVersion2]("test_index_update_type")
 
-  it should "create an index and update the index field types" in {
-    val indexVersion1 = ElasticsearchIndex(
-      "readgroups1",
-      "default",
-      Seq(ElasticsearchField("bar", classOf[String]))
-    )
-    val indexVersion2 = ElasticsearchIndex(
-      "readgroups1",
-      "default",
-      Seq(
-        ElasticsearchField("foo", classOf[Long]),
-        ElasticsearchField("bar", classOf[String])
-      )
-    )
     for {
-      _ <- httpElasticsearchDAO.createIndexType(
-        indexVersion1,
-        replicate = false
-      )
+      _ <- httpElasticsearchDAO.createIndexType(indexVersion1)
```

Comment:
Yup.

`<-` is a generator.

`=` is a definition.

http://www.artima.com/pins1ed/for-expressions-revisited.html#23.1